client/utils.go

```diff
@@ -0,0 +1,49 @@
+/*************************************************************************
+ * Copyright 2023 Gravwell, Inc. All rights reserved.
+ * Contact: <legal@gravwell.io>
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD 2-clause license. See the LICENSE file for details.
+ **************************************************************************/
+
+package client
+
+import (
+	"errors"
+	"io"
+	"net/http"
+)
+
+const (
+	maxDataDrain = 1024 * 1024 * 4
+)
+
+var (
+	ErrMaxBodyDrained = errors.New("too many response bytes in body, closing")
+)
+
+// drainResponse will drain up to 4MB of data then close the response Body.
+// We do this so that http requests can re-use connects as per docs.
+func drainResponse(resp *http.Response) {
+	if resp.Body == nil {
+		return
+	}
+	var nw nilWriter
```

Comment:
why not use ioutil.Discard as the writer here?

Reply:
a limitedWriter + io.Discard would be equivalent.  This felt just as simple.